### PR TITLE
feat: @contentful/f36-card

### PR DESCRIPTION
**Pull request by denkristoffer**
# Purpose of PR

This PR creates `@contentful/f36-card` with the components `Card`, `AssetCard`, `EntryCard`, and `InlineEntryCard` with new design. After considering different approaches I built this with a `BaseCard` component that can show a custom header component, as this seemed the best way to base all cards on the same foundational component to share props and styling as much as possible. 

Follow-up PRs:

- Loading state with skeletons for all cards.

### CODE REVIEW

File: packages/components/card/src/base-card/CardActions.tsx
Comment (suevalov) [inline packages/components/card/src/base-card/CardActions.tsx:6]: shouldn't we wait till the Dropdown is migrated to be able to extract card to a separate package?

File: packages/components/card/src/card/README.mdx
Comment (suevalov) [inline packages/components/card/src/card/README.mdx:29]: `Typograpy` wrapper can be deleted

File: packages/components/card/package.json
Comment (suevalov) [inline packages/components/card/package.json:22]: we need to remove this line before the official release.

**Comment by suevalov** [comment]
In general, I like the idea of the BaseCard component that handles commonalities.

I guess we need to wait for the Dropdown component before merging this PR and also work on types a bit (so `href` becomes possible only when you specify <Card as="a" /> for example)

File: packages/components/card/src/base-card/CardActions.tsx
Comment (denkristoffer) [reply packages/components/card/src/base-card/CardActions.tsx:6]: I don't think we have to, but we can if you think that's better.

**Comment by denkristoffer** [comment]
> and also work on types a bit (so href becomes possible only when you specify for example)

That's should be how it works now with https://github.com/contentful/forma-36/pull/1114/files#diff-73ea3ba88ccb975ddf6b2bc2987806c1cafa584382271a11cf23a392b192e149R69-R91 🙂

**Comment by suevalov** [comment]
@denkristoffer `Menu` component is coming, so we hope to unblock this PR this Sprint.

**Comment by suevalov** [comment]
@denkristoffer I've pushed the changes that use the Menu component instead of the Dropdown. We can finally focus on making PR ready for a review.

File: packages/components/card/src/asset-card/AssetCard.styles.ts
Comment (suevalov) [inline packages/components/card/src/asset-card/AssetCard.styles.ts:23]: It would be nice to repalce all properties that use calculated spacing value (using fontBaseDefault) with the closest spacing token.

File: packages/components/card/src/asset-card/AssetCard.tsx
Comment (suevalov) [inline packages/components/card/src/asset-card/AssetCard.tsx:24]: Can we use spacing value here as well?

File: packages/components/card/src/asset-card/AssetCard.tsx
Comment (suevalov) [inline packages/components/card/src/asset-card/AssetCard.tsx:57]: do we really need to allow overriding `actionButtonProps`?

File: packages/components/card/src/base-card/BaseCard.tsx
Comment (suevalov) [inline packages/components/card/src/base-card/BaseCard.tsx:69]: I think these properties can be automatically recognized when you use `as="a"`. I'm not sure if you need to specify this type here.

File: packages/components/card/src/base-card/CardActions.tsx
Comment (suevalov) [inline packages/components/card/src/base-card/CardActions.tsx:17]: do we use this property anywhere?

File: packages/components/card/src/card/__snapshots__/Card.test.tsx.snap
Comment (suevalov) [inline packages/components/card/src/card/__snapshots__/Card.test.tsx.snap:3]: It would be nice to delete this snapshot test.

File: packages/components/card/src/asset-card/AssetCard.styles.ts
Comment (denkristoffer) [reply packages/components/card/src/asset-card/AssetCard.styles.ts:23]: @fabe @domarku can we do that? 🙂

File: packages/components/card/src/base-card/CardActions.tsx
Comment (denkristoffer) [reply packages/components/card/src/base-card/CardActions.tsx:17]: It's used by the `InlineEntryCard`

File: packages/components/card/src/asset-card/AssetCard.tsx
Comment (domarku) [reply packages/components/card/src/asset-card/AssetCard.tsx:24]: This 56px is just for the Storybook preview, correct?

File: packages/components/card/src/asset-card/AssetCard.styles.ts
Comment (domarku) [reply packages/components/card/src/asset-card/AssetCard.styles.ts:26]: Yes! Let's use `spacing-xs` here. It will reduce the height of the header section to `36px`, which is a good thing because it will contribute to items taking less vertical real-estate.

The only downside is that the "more" button's height will go down `4px`. I don't think that's terrible as it will still be close to `40px` in the size of its click-area.

File: packages/components/card/src/asset-card/AssetCard.tsx
Comment (suevalov) [inline packages/components/card/src/asset-card/AssetCard.tsx:23]: can we use spacing tokens here?

File: packages/components/card/src/card/__snapshots__/Card.test.tsx.snap
Comment (suevalov) [inline packages/components/card/src/card/__snapshots__/Card.test.tsx.snap:3]: We decided to delete snapshot tests, so this file should be gone, together with a corresponding test.

**Review by suevalov — APPROVED** [review]
👏🏼 

Only two small comments

File: packages/core/src/utils/getEntityStatusStyles.ts
Comment (gui-santos) [inline packages/core/src/utils/getEntityStatusStyles.ts:35]:
```suggestion
      console.warn(`Unknown EntityStatus: ${status}`);
```

Maybe warn? So it show very nice yellow message in the console 👌

File: packages/components/card/src/inline-entry-card/InlineEntryCard.tsx
Comment (gui-santos) [inline packages/components/card/src/inline-entry-card/InlineEntryCard.tsx:31]: Shouldn't we move all these `getStyles` functions to the `.styles.ts` files?

**Review by gui-santos — APPROVED** [review]
💎 

I have only two small suggestions that are not blocking this to be merged tbh

**Review by thebesson — APPROVED** [review]

File: packages/components/card/src/inline-entry-card/InlineEntryCard.tsx
Comment (denkristoffer) [reply packages/components/card/src/inline-entry-card/InlineEntryCard.tsx]: Yeah. Changed to a nice pattern I picked up from @bgutsol 👍

File: packages/components/card/src/asset-card/AssetCard.tsx
Comment (denkristoffer) [reply packages/components/card/src/asset-card/AssetCard.tsx]: I'm discussing this with @domarku

**Comment by denkristoffer** [comment]
Merging, anything else can be done as a follow-up.

**Pull request merged**
